dcp/DcpClient.go

```diff
@@ -210,14 +211,21 @@ func (c *DcpClient) initializeCluster() (err error) {
 			return err
 		}
 	}
+
 	return nil
 }
 
 func initializeClusterWithSecurity(dcpDriver *DcpDriver) (*gocb.Cluster, error) {
 	clusterOpts := gocb.ClusterOptions{}
 
-	if dcpDriver.ref.HttpAuthMech() == xdcrBase.HttpAuthMechHttps {
-		tlsCert := tls.Certificate{Certificate: [][]byte{dcpDriver.ref.Certificate()}}
+	useCouchbaseSecureStr := dcpDriver.ref.HttpAuthMech() == xdcrBase.HttpAuthMechHttps
+
+	// If it is a source cluster, use cbauth username/pw and not client certs
+	if dcpDriver.Name != base.SourceClusterName && len(dcpDriver.ref.ClientCertificate()) > 0 && len(dcpDriver.ref.ClientKey()) > 0 {
+		tlsCert := tls.Certificate{
+			Certificate: [][]byte{dcpDriver.ref.Certificate()},
+			PrivateKey:  dcpDriver.ref.ClientKey(),
```

Comment:
hi Neil, from the `master` branch it seems like `dcpDriver.ref.ClientKey()` essentially returns a byte array of the client key stored in the `RemoteClusterReference` object

however the `tls.Certificate` expects a `crypto.PrivateKey` type value, which needs to implement certain interface methods

do we need to use `tls.X509KeyPair()` instead to generate the `tls.Certificate`? (assuming we only use x509 certs)

@sumukhbhat2701 @darshan-na do you know if the cluster reference methods `Certificate()` and `ClientKey()` return the relevant PEM blocks?

Reply:
from the `crypto/tls` docs:

> // X509KeyPair parses a public/private key pair from a pair of
// PEM encoded data. On successful return, Certificate.Leaf will be nil because
// the parsed form of the certificate is not retained.
func X509KeyPair(certPEMBlock, keyPEMBlock []byte) (Certificate, error) {

Reply:
focus is on username/password TLS, marking this as done